### ISSUE TITLE
config validation: reconcile the two config validation paths

### ISSUE THREAD

**Issue by ccaraman** [envoyproxy/envoy#2636]
The envoy binary supports a validate mode to check the validity of a configuration and there is a config check tool that does the same.  The config check tool should be updated to use the Validation mode in Server. 

Server Validation https://github.com/envoyproxy/envoy/blob/master/source/server/config_validation/server.cc

Config check tool
https://github.com/envoyproxy/envoy/blob/master/test/config_test/config_test.cc

**Comment by htuch** [comment]
+1 I think having convergence here would be great, if only to ensure we're getting burn time on the server validation path; it has some significant initialization reimplementation, so good to keep it in active use.

**Comment by akonradi** [comment]
I've started working on this and run into a few snags. It looks like the existing config validation tool wasn't checking the Envoy runtime layers, so when attempting to load them now in `validate` mode, it complains about not being able to watch symlinks. I'm not quite clear on what the semantics of the config check tool are. Should it check that all requisite files and directories are present? Or would it be sufficient to use the proposed `load` mode in https://github.com/envoyproxy/envoy/issues/9809#issuecomment-782426767, which doesn't assume as much about the file system?

**Comment by mattklein123** [comment]
Thanks @akonradi for looking into this. Does the existing config check tool actually try to load TLS secrets from the filesystem? If so I think I would also expect it to actually check for the symlinks, so IMO that is a bug?

I agree that we shouldn't special case TLS secrets. Either it loads from the FS or it doesn't.

**Comment by akonradi** [comment]
The existing config check tool does not, AFAICT, attempt to load TLS secrets from the file system, otherwise it would fail on some of the example configs. I've realized that what I attempted in #15158 for `//test/config_test` has already been done more cleanly in `//test/server/config_validation:server_test`, except that it has a TODO about not being able to validate all example configs since they depend on file-system resources. I'm going to start there and work on adding `Mode::load` and support, and plan to refactor `//test/config_test` into a very thin wrapper around the same code.
